lib/asset_cloud/base.rb

```diff
@@ -8,8 +8,25 @@ class IllegalPath < StandardError
 
   class Base
     cattr_accessor :logger
-
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A
+      (
+        ([a-z0-9])            #Filename can be a single letter
+        |                     #OR it is many and follows the below rules
+        (
+          (\.?[\w\-\@])       #It can start with a dot but it must have a following character
+          (
+            [\w\-\@]          #You can have a letter without any following conditions
+            |
+            [\ ][\w\-\@]      #If there is a space you need to have a normal letter afterward
+            |
+            [\/][\w\-\@]      #If there is a slash you need to have a normal letter afterward
+            |
+            [\/][\.][\w\-\@]  #Though a slash could be followed by a dot so long as there is a normal letter afterward
+            |
+            [\.][\w\-\@]+     #A dot must be followed by one (or more) normal letters
+          )*                  #Zero to many of these combinations.
+        )
+      )\z/ix                   #Case insensitive
```

Comment:
Heh, yeah -- heh, i didn't intend to pawn that quote off as my own: http://regex.info/blog/2006-09-15/247